sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala

```diff
@@ -190,10 +190,15 @@ class DataFrameReader private[sql](sparkSession: SparkSession) extends Logging {
     val cls = DataSource.lookupDataSource(source, sparkSession.sessionState.conf)
     if (classOf[DataSourceV2].isAssignableFrom(cls)) {
       val ds = cls.newInstance()
-      val options = new DataSourceOptions((extraOptions ++
-        DataSourceV2Utils.extractSessionConfigs(
-          ds = ds.asInstanceOf[DataSourceV2],
-          conf = sparkSession.sessionState.conf)).asJava)
+      val sessionOptions = DataSourceV2Utils.extractSessionConfigs(
+        ds = ds.asInstanceOf[DataSourceV2],
+        conf = sparkSession.sessionState.conf)
+      val pathOption = if (paths.isEmpty) {
+        Map.empty
+      } else {
+        Map(DataSourceOptions.KEY_PATH -> paths.mkString(","))
```

Comment:
`KEY_PATH` sounds like the path for a key. It would be more clear if the name was `PATH_KEY`.